docs/DesignDocs/SGX2-EDMM-driver-interface.md

```diff
@@ -0,0 +1,170 @@
+SGX EDMM Driver Interface Design
+=====================================
+
+## Motivation
+
+This document describes possible driver interfaces to facilitate discussions among SGX runtime implementors (e.g., https://github.com/openenclave/openenclave/pull/3639) on supporting different SGX EDMM flows.
+
+Although interfaces described here are inspired to be as likely as possible a candidate for future kernel adoption, they are not intended to be a proposal for kernel implementation and are assumed to be implemented as an OOT driver. We hope from discussions enabled by this document, requirements and usage models can be identified to help shape future kernel interfaces. 
```

Comment:
Regarding "kernel" adoption, is this applicable to any OS (Linux, Windows, etc.) or only Linux?  Clarify.

Reply:
clarified in title and introduction